### ISSUE TITLE
Error when running ms3.isolation_metrics

### ISSUE THREAD

**Issue by khl02007** [flatironinstitute/mountainsort#28]
**Hardware**: Amazon EC2 instance (c4.8xlarge, Ubuntu 16.04, 200GB storage space)
**Installation**:  from ppa:magland/mountainlab via apt on Ubuntu 
**Pipeline**: mountainsort3.mlp from mountainsort_examples
**Data**: silicon probe recording, ~ 20GB with ~ 100 channels
**Issue**: filtering, whitening, and sorting runs fine, but curation returns an error while running ms3.isolation_metrics
**Note**: I was able to sort the test data from mountainsort_examples without any problem. Numpy was also upgraded with `pip3 install numpy`
**Error message**:
```
----------------------------------------------------------------------------
Queueing job: ms3.cluster_metrics
  INPUTS: timeseries=[object Object]  firings=[object Object]
  PARAMS: samplerate=20000
----------------------------------------------------------------------------
----------------------------------------------------------------------------
Queueing job: ms3.isolation_metrics
  INPUTS: timeseries=[object Object]  firings=[object Object]
  PARAMS: compute_bursting_parents=true
----------------------------------------------------------------------------
  |ms3.cluster_metrics|            RUNNING ms3.cluster_metrics: /opt/mountainlab/packages/ms3/ms3.mp ms3.cluster_metrics --firings=/home/ubuntu/test/home/ubuntu/tmp/mountainlab/prvbucket/_mountainprocess/output_firings_out_20e3ed3ea5 --timeseries=/home/ubuntu/test/home/ubuntu/tmp/mountainlab/prvbucket/_mountainprocess/output_timeseries_out_67c7c902ec --cluster_metrics_out=/home/ubuntu/test/home/ubuntu/tmp/mountainlab/prvbucket/_mountainprocess/output_cluster_metrics_out_189e344a22 --samplerate=20000 --_tempdir=/home/ubuntu/test/home/ubuntu/tmp/mountainlab/tmp_short_term/tempdir_0XT36758GF
  |ms3.isolation_metrics|          RUNNING ms3.isolation_metrics: /opt/mountainlab/packages/ms3/ms3.mp ms3.isolation_metrics --firings=/home/ubuntu/test/home/ubuntu/tmp/mountainlab/prvbucket/_mountainprocess/output_firings_out_20e3ed3ea5 --timeseries=/home/ubuntu/test/home/ubuntu/tmp/mountainlab/prvbucket/_mountainprocess/output_timeseries_out_67c7c902ec --metrics_out=/home/ubuntu/test/home/ubuntu/tmp/mountainlab/prvbucket/_mountainprocess/output_metrics_out_efa29395f3 --compute_bursting_parents=true --_tempdir=/home/ubuntu/test/home/ubuntu/tmp/mountainlab/tmp_short_term/tempdir_D07K11PRZU
  |ms3.isolation_metrics|          Starting p_isolation_metrics
  |ms3.isolation_metrics|          Computing cluster metrics...
  |ms3.cluster_metrics|            Process completed successfully: ms3.cluster_metrics (Elapsed: 7.568 sec)
  |ms3.cluster_metrics|            making prv file: /home/ubuntu/test/home/ubuntu/tmp/mountainlab/prvbucket/_mountainprocess/output_cluster_metrics_out_189e344a22.prv
  |ms3.cluster_metrics|            making prv file: /home/ubuntu/test/home/ubuntu/tmp/mountainlab/prvbucket/_mountainprocess/output_console_out_189e344a22.prv
  |ms3.isolation_metrics|          2018-01-31:23-14-39-064      mproc.main      warning
  |ms3.isolation_metrics|          2018-01-31:23-14-39-072      mproc.main      warning Process returned with non-zero exit code (9): ms3.isolation_metrics (Elapsed: 22.806 sec)
  |ms3.isolation_metrics|          making prv file: /home/ubuntu/test/home/ubuntu/tmp/mountainlab/prvbucket/_mountainprocess/output_console_out_efa29395f3.prv
Process error:
Process error:
Error in job (processor: ms3.isolation_metrics): Process error:
Could not find input in running or pending jobs: metrics2
Error in job (pipeline: curate): Error in job (processor: ms3.isolation_metrics): Process error:
Stopped
Stopped
Stopped
Stopped
Stopped
Stopped
Error: Error in job (pipeline: curate): Error in job (processor: ms3.isolation_metrics): Process error:
Error: Error in job (pipeline: curate): Error in job (processor: ms3.isolation_metrics): Process error:
```

**Comment by magland** [comment]
Thanks for the detailed report. How much RAM?

I suspect it is running out of memory. It is a known problem that the
computation of isolation metrics needs to be improved to not use so much
RAM.

**Comment by khl02007** [comment]
I see. I tried running the same data set again on a different Amazon EC2 instance with larger memory (m4.16xlarge with 64 vCPUs and 256 GiB of memory) and it worked fine - thank you. I also noticed that the CPU utilization is low when computing isolation metrics. Is there a plan to parallelize it so that it can be run on multiple cores simultaneously? For my data set, computing the isolation metrics takes much longer than the sorting part.
